src/AuthorizedSocketConnection.ts

```diff
@@ -185,14 +194,15 @@ export default class AuthorizedSocketConnection<TContext, TCredentials> {
         null,
         {
           subscribe: async (
-            topic,
+            topic: string,
             {
               hasPermission = this.config.hasPermission,
               ...options
             }: SubscribeOptions<TCredentials> = {},
           ) => {
-            return AsyncUtils.filter(
-              await subscriptionContext.subscribe(topic, options),
+            return AsyncUtils.filter<any>(
+              // FIXME: type this correctly
+              (await subscriptionContext.subscribe(topic, options)) as any,
```

Comment:
some of `AsyncUtils` accept an `AsyncIterable` as the first param (which makes sense) but ts was erroring b/c we're passing in `AsyncIterator`s?

Reply:
okay, i took a look. my typing was wrong. we should be passing around `AsyncIterable`s everywhere, and not `AsyncIterator`s. `for await` wants an `AsyncIterable`. it just so happens that `AsyncGenerator`s are both `AsyncIterable`s and `AsyncIterator`s